benchmarks/main.cpp

```diff
@@ -0,0 +1,284 @@
+#include "BenchmarkBase.h"
+
+using namespace std;
+
+// Define the global benchmark results storage
+map<string, BenchmarkResult> g_benchmarkResults;
+
+// ANSI color codes for terminal output
+const string GREEN = "\033[32m";
+const string RED = "\033[31m";
+const string RESET = "\033[0m";
+const string BOLD = "\033[1m";
+
+bool runBenchmarks(const set<string>& include, const set<string>& exclude) {
+    g_benchmarkResults.clear();
+    int result = tpunit::Tests::run(include, exclude, {}, {}, 1);
+    return result == 0;
+}
+
+bool executeCommand(const string& cmd, string* output = nullptr) {
+    string fullCmd = cmd;
+    if (output) {
+        fullCmd += " 2>&1";  // Capture stderr too when output is requested
+    }
+
+    FILE* pipe = popen(fullCmd.c_str(), "r");
+    if (!pipe) {
+        return false;
+    }
+
+    char buffer[256];
+    string result;
+    while (fgets(buffer, sizeof(buffer), pipe) != nullptr) {
+        result += buffer;
+        if (!output) {
+            cout << buffer;  // Stream output in real-time if not capturing
+        }
+    }
+
+    if (output) {
+        *output = result;
+    }
+
+    int status = pclose(pipe);
+    // pclose returns -1 on error, otherwise the exit status
+    if (status == -1) {
+        return false;
+    }
+
+    // On success, WEXITSTATUS should be 0
+    return WIFEXITED(status) && WEXITSTATUS(status) == 0;
+}
+
+void printComparison(const map<string, BenchmarkResult>& baseline,
+                     const map<string, BenchmarkResult>& current) {
+    cout << "\n" << BOLD << "=== Benchmark Comparison Report ===" << RESET << "\n\n";
+    
+    // Find the longest benchmark name for formatting
+    size_t maxNameLen = 0;
+    for (const auto& [name, _] : current) {
+        maxNameLen = max(maxNameLen, name.length());
+    }
+    maxNameLen = max(maxNameLen, size_t(20)); // Minimum width
+    
+    // Print header
+    cout << left << setw(maxNameLen + 2) << "Benchmark"
+         << right << setw(15) << "Baseline MB/s"
+         << setw(15) << "Current MB/s"
+         << setw(12) << "Change %"
+         << setw(15) << "Status" << "\n";
+    cout << string(maxNameLen + 57, '-') << "\n";
+    
+    // Compare results
+    for (const auto& [name, currentResult] : current) {
+        auto it = baseline.find(name);
+        if (it == baseline.end()) {
+            // New benchmark
+            cout << left << setw(maxNameLen + 2) << name
+                 << right << setw(15) << "N/A"
+                 << setw(15) << fixed << setprecision(2) << currentResult.throughputMBps
+                 << setw(12) << "NEW"
+                 << setw(15) << "✨ NEW" << "\n";
+        } else {
+            const auto& baselineResult = it->second;
+            double changePercent = ((currentResult.throughputMBps - baselineResult.throughputMBps) 
+                                   / baselineResult.throughputMBps) * 100.0;
+            
+            // Use colors for significant changes (>1% difference)
```

Comment:
1% seems kinda low, no? I'd imagine something like 5-10% being a better indicator of performance change